### PR TITLE
Temperature regression vignette

### PR DESCRIPTION
**Pull request by B-yunes**
- Used water temperature regression markdown as a starting point / reference
- Created `temp_regression_vignette.Rmd` 
- Added information to explain processes of water temperature regression 
-  Cleaned markdown up to be added to the environmental covariates vignette

### CODE REVIEW

File: data-raw/temperature-data/temp_regression_vignette.Rmd [hunk 1]
```diff
@@ -0,0 +1,797 @@
...
+test_predict_df <- test |>
+  mutate(predicted = test_predict)
+# evaluate predictions - MAPE of 10% is not bad
+mean(abs((
+  test_predict_df$predicted - test_predict_df$temp
+)) / test_predict_df$temp)
+```
+
+#### Building Min Regression
+
```
Comment (alvizek) [inline data-raw/temperature-data/temp_regression_vignette.Rmd:249]: Lets work on formatting these sections a little more cleanly. 

The first part of the code sets up and runs the model so you could show that output with some text and then after that chunk include the output of evaluating the model. I don't think we need to show all the code for that and instead can include what the calculated mean absolute error was (might want to look that up to learn what it tells you so can explain it in the text).

This applies for all the min/mean/max sections

File: data-raw/temperature-data/temp_regression_vignette.Rmd [hunk 2]
```diff
@@ -0,0 +1,797 @@
...
+  select(date, value, statistic)
+ggplot(feather_lfc_max_predicted, aes(x = date, y = value)) +
+  geom_line()
+```
+
+#### Full dataset
+
+  * Join with original data (merge combined predictions with the original dataset that includes gage agency and gage number)
+  * Visualize combined data
+
```
Comment (alvizek) [inline data-raw/temperature-data/temp_regression_vignette.Rmd:330]: I don't think we need to show all this code.

Also should include some interpretation of the plots

File: data-raw/temperature-data/temp_regression_vignette.Rmd [hunk 3]
```diff
@@ -0,0 +1,797 @@
...
+mean(abs((
+  test_predict_df$predicted - test_predict_df$temp
+)) / test_predict_df$temp)
+```
+
+#### Predictions
+
+  * Combine Predictions (mean, min, max dataframes into one)
+  * Reshape data
+
```
Comment (alvizek) [inline data-raw/temperature-data/temp_regression_vignette.Rmd:300]: I don't think we need to show all this code. Include interpretation of plots.

**Review by alvizek — CHANGES_REQUESTED** [review]
I made some edits directly to the rmd and then ran out of time. I pushed my changes. I also added some comments. I think more work needs to be done to explain the plots and results that you are including in the Rmd.

File: data-raw/temperature-data/temp_regression_vignette.Rmd [hunk 4]
```diff
@@ -0,0 +1,797 @@
+---
```
[the remark this answers is not in the document]
Comment (B-yunes) [reply data-raw/temperature-data/temp_regression_vignette.Rmd:1]: Changed the title to this suggestion

File: data-raw/temperature-data/CopyOfwater_temp_regression.html
[the remark this answers is not in the document]
Comment (B-yunes) [reply data-raw/temperature-data/CopyOfwater_temp_regression.html:1]: deleted file

**Review by alvizek — APPROVED** [review]

**Pull request merged**